paper-icon-button-light.html

```diff
@@ -0,0 +1,129 @@
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+<link rel="import" href="../paper-behaviors/paper-ripple-behavior.html">
+
+<!--
+This is a lighter version of `paper-icon-button`. Its goal is performance, not
+developer ergonomics, so as a result it has fewer features than `paper-icon-button`
+itself. To use it, you must distribute a `button` containing the `iron-icon` you
+want to use:
+
+<link rel="import" href="../iron-icon/iron-icon.html">
+<link rel="import" href="../paper-icon-button/paper-icon-button-light.html">
+<link rel="import" href="../iron-icons/iron-icons.html">
+
+<paper-icon-button-light>
+  <button title="heart">
+    <iron-icon icon="favorite"></iron-icon>
+  </button>
+</paper-icon-button-light>
+
+Note that this button is assumed to be distributed at the startup of
+`paper-icon-button-light`. Dynamically adding a `button` to this element is
+not supported.
+
+The `title`/`disabled` etc. attributes go on the distributed button, not on the wrapper.
+
+The following custom properties and mixins are also available for styling:
+Custom property | Description | Default
+----------------|-------------|----------
+`--paper-icon-button-light-ripple` | Mixin applied to the paper ripple | `{}`
+
+@group Paper Elements
+@element paper-icon-button-light
+@demo demo/paper-icon-button-light.html
+-->
+
+<dom-module id="paper-icon-button-light">
+  <template strip-whitespace>
+    <style>
+      :host {
+        display: inline-block;
+        position: relative;
+        width: 24px;
+        height: 24px;
+      }
+
+      paper-ripple {
+        opacity: 0.6;
+        color: currentColor;
+        @apply(--paper-icon-button-light-ripple);
+      }
+
+      :host ::slotted(button) {
+        position: relative;
+        width: 100%;
+        height: 100%;
+        margin: 0;
+        padding: 0;
+        background: none;
+        border: none;
+        outline: none;
+        vertical-align: middle;
+        color: inherit;
+        cursor: pointer;
+        /* NOTE: Both values are needed, since some phones require the value to be `transparent`. */
+        -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+        -webkit-tap-highlight-color: transparent;
+      }
+      :host ::slotted(button[disabled]) {
+        color: #9b9b9b;
+        pointer-events: none;
+        cursor: auto;
+      }
+    </style>
+    <slot id="content"></slot>
+  </template>
+
+  <script>
+    Polymer({
+      is: 'paper-icon-button-light',
+
+      behaviors: [
+        Polymer.PaperRippleBehavior
+      ],
+
+      listeners: {
+        'down': '_rippleDown',
+        'up': '_rippleUp'
+      },
+
+      attached: function() {
+        // Assume the button has already been distributed.
+        var button = this.getEffectiveChildren()[0];
```

Comment:
Is this safe if the <paper-icon-button-light> is created during parsing? I think connectedCallback might be called before the element's children are parsed / appended in that case.

Reply:
moved to `ready()`